### PR TITLE
Deploy the website based on the new docs and samples

### PR DESCRIPTION
**Pull request by simonbrunel**
- remove old docs (and task) - it's now deployed from the Chart.js repo (chartjs/Chart.js#4256)
- add Travis CI task to deploy the website to chartjs.github.io for the `release` branch
- chartjs.org/docs now redirects to the latest version
- chartjs.org/samples redirects to the latest version
- update homepage links ("Get Started", "Samples", "GitHub")
- use relative paths for internal links
- enable ESLint

Fixes chartjs/Chart.js#2383

image

> **Note**
> - "Get Started" redirects to https://www.chartjs.org/docs
> - "Samples" redirects to https://www.chartjs.org/samples
> - "GitHub" redirects to https://github.com/chartjs/Chart.js

### CODE REVIEW

File: scripts/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -e
+
+TARGET_DIR='gh-pages'
+TARGET_BRANCH='master'
+TARGET_REPO_URL="https://$GITHUB_AUTH_TOKEN@github.com/chartjs/chartjs.github.io.git.git"
```
Comment (benmccann) [inline scripts/deploy.sh]: is this line supposed to end in `.git.git`? I would have guessed you'd have `.git` only once

**Comment by benmccann** [comment]
lgtm

File: scripts/deploy.sh (hunk 1 above)
Comment (simonbrunel) [reply scripts/deploy.sh]: very good catch, thanks!

**Review by etimberg — APPROVED** [review]

**Comment by benmccann** [comment]
@nnnick do these changes look okay to you?

**Review by nnnick — APPROVED** [review]
Looks great! Good job 🎉

**Pull request merged**